lib/instances/Script.lua

```diff
@@ -9,6 +9,41 @@ local Script = BaseInstance:extend("Script", {
 	creatable = true,
 })
 
+function Script:init(instance)
+	local internal = getmetatable(instance).instance
+
+	internal.hasRun = false
+
+	instance.AncestryChanged:Connect(function()
+		if internal.hasRun then
+			return
+		end
+
+		local game = nil
+
+		local now = instance
+		while now do
```

Comment:
What's the point of this whole loop? If you really can't just use the game global, changing this to now:FindFirstAncestorWhichIsA("DataModel") would make this more readable.

Reply:
Right, there is no `game` global here, since we have no environment.

I forgot that `FindFirstAncestorWhichIsA` was a function we have, that's the way to go. I'll remove the existing ScreenGui checks and replace them with that too.

Reply:
On second thought, FindFirstAncestorWhichIsA isn't actually implemented on lemur. Wouldn't be hard though.